test/unit/fact_cleaner_test.rb

```diff
@@ -50,11 +50,7 @@ class FactCleanerTest < ActiveSupport::TestCase
   end
 
   test "it cleans excluded facts" do
-    name_generator_type = Setting[:name_generator_type]
-    host_owner = Setting[:host_owner]
-    Setting.stubs(:[]).with(:excluded_facts).returns(['ignored*', '*bad'])
-    Setting.stubs(:[]).with(:name_generator_type).returns(name_generator_type)
-    Setting.stubs(:[]).with(:host_owner).returns(host_owner)
+    Setting[:excluded_facts] = ['ignored*', '*bad']
```

Comment:
This still has a potential to influence other tests, that would expect original setting. What about using `unstub` in `teardown` instead?

Reply:
Or restore original value after the test, e.g.
```ruby
a,b='new',a
$test
a=b
```

Reply:
I think the database is isolated between tests anyway. If we add 
``` ruby 
Rails.application.configure do
  config.cache_store = :null_store
end
```

to the mix, settings should be isolated completely.